vllm/v1/engine/async_llm.py

```diff
@@ -56,11 +56,14 @@ def __init__(
 
         self.log_requests = log_requests
         self.log_stats = log_stats
-        self.stat_loggers: list[StatLoggerBase] = []
+        self.stat_loggers: dict[str, StatLoggerBase] = dict()
         if self.log_stats:
-            if logger.isEnabledFor(logging.INFO):
-                self.stat_loggers.append(LoggingStatLogger())
-            self.stat_loggers.append(PrometheusStatLogger(vllm_config))
+            if stat_loggers is not None:
+                self.stat_loggers = stat_loggers
+            else:
+                setup_default_loggers(vllm_config,
+                                      logger.isEnabledFor(logging.INFO),
+                                      self.stat_loggers)
```

Comment:
Suggest moving all of this logic into the setup function:
```python
self.stat_loggers = setup_default_loggers(vllm_config, log_stats, stat_loggers)
```

Also:
- No need to pass in `logger.isEnabledFor(logging.INFO)`, that can be done inside the function
- It would probably be cleaner to eliminate `self.log_stats`. Empty `self.stat_loggers` can serve that purpose (or possibly `None`)